packages/core/src/api/blockManipulation/tables/tables.ts

```diff
@@ -224,78 +388,344 @@ export function resolveAbsoluteTableCellIndices(
+ *
+ * // getCellsAtColumnHandle(1)
+ * // returns [
+ *  { row: 0, col: 1, cell: "B" },
+ *  { row: 1, col: 2, cell: "E" },
+ *  { row: 2, col: 2, cell: "F" },
+ * ]
+ * ```
+ *
+ * As you can see, you may not be able to retrieve all nodes given a relative column index, as cells can span multiple columns.
+ *
+ * @returns All of the cells associated with the relative column of the table. (All cells that have the same relative column index)
  */
-export function getColumn(
+export function getCellsAtColumnHandle(
   block: BlockFromConfigNoChildren<DefaultBlockSchema["table"], any, any>,
-  relativeColumnIndex: number
+  relativeColumnIndex: RelativeCellIndices["col"]
 ) {
-  const occupancyGrid = getTableCellOccupancyGrid(block);
-  if (!occupancyGrid) {
-    throw new Error("Table is malformed");
+  try {
+    const occupancyGrid = getTableCellOccupancyGrid(block);
+    // First need to resolve the relative column index to an absolute column index
+    let absoluteCol = 0;
+
+    // Now that we've already resolved the absolute row position, we can jump through the occupied cells ${relativeCellIndices.col} times to find the absolute column position
+    for (let i = 0; i < relativeColumnIndex; i++) {
+      const cell = occupancyGrid[0]?.[absoluteCol];
+
+      if (!cell) {
+        // As a sanity check, if the cell is not occupied, we should throw an error
+        throw new Error(
+          `Unable to resolve relative table cell indices for table, cell at 0,${absoluteCol} is not occupied`
+        );
+      }
+
+      // Skip the cells that the colspan takes up
+      absoluteCol += cell.colspan;
+    }
+
+    // Then for each row, get the cell at the absolute column index as a relative cell index
+    const cells = new Array(occupancyGrid.length).fill(false).map((_v, row) => {
+      return getRelativeTableCellIndices(
+        { row, col: absoluteCol },
+        block,
+        occupancyGrid
+      );
+    });
+
+    // Filter out duplicates based on row and col properties
+    return cells.filter((cell, index) => {
+      return (
+        cells.findIndex((c) => c.row === cell.row && c.col === cell.col) ===
+        index
+      );
+    });
+  } catch (e) {
+    // In case of an invalid index, return an empty array
+    return [];
   }
-  const { height } = getDimensionsOfTable(block);
-  // First need to resolve the relative column index to an absolute column index
-  const { col: absoluteCol } = resolveRelativeTableCellIndices(
-    { row: 0, col: relativeColumnIndex },
-    block
+}
+
+/**
+ * This moves a column from one index to another.
+ *
+ * @note This is a destructive operation, it will modify the {@link OccupancyGrid} in place.
+ */
+export function moveColumn(
+  block: BlockFromConfigNoChildren<DefaultBlockSchema["table"], any, any>,
+  fromColIndex: RelativeCellIndices["col"],
+  toColIndex: RelativeCellIndices["col"],
+  occupancyGrid: OccupancyGrid = getTableCellOccupancyGrid(block)
+): TableContent<any, any>["rows"] {
```

Comment:
as discussed; let's add a comment that it would be nice to throw an error when one of the cells has a colspan or rowspan (no need to actually implement the error at this point imo)